PlayerCollection.cs

```diff
@@ -93,9 +109,19 @@ private CommonFields.Player GetArcPlayer(string characterName)
             return _arcPlayersInSquad.First(x => x.Value.CharacterName.Equals(characterName)).Value;
         }
 
-        private PlayerDisplay GetPlayer(CommonFields.Player arcPlayer)
+        private bool TryGetPlayer(CommonFields.Player arcPlayer, out PlayerDisplay playerDisplay)
         {
-            return _playerDisplays.First(x => x.AccountName.Equals(arcPlayer.AccountName));
+            try
+            {
+                playerDisplay = _playerDisplays.First(x => x.AccountName.Equals(arcPlayer.AccountName));
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Logger.Error($"Caught exception while trying to get PlayerDisplay to remove from Current Squad Members: {ex}");
+                playerDisplay = null;
+                return false;
+            }
```

Comment:
No need to use a try/catch block, you can replace `First` by `FirstOrDefault`, which returns null instead of throwing when nothing has been found

Reply:
I've pushed a change for this - couldn't see a need to keep the logging statement. Anything you'd log here?